src/test/java/com/arkvis/orchid/CalendarTest.java

```diff
@@ -105,4 +106,47 @@ void should_returnCorrectFlow_when_periodAddedWithSpottingFlow(){
 
         assertEquals(Flow.SPOTTING, retrievedPeriod.getFlow());
     }
+
+    @Test
+    void should_returnTemperature_when_temperatureAddedToDay(){
+        LocalDate date = LocalDate.now();
+        PeriodCalendar calendar = new PeriodCalendar(new PeriodPredictor());
+
+        Temperature temperature = new Temperature(new BigDecimal(100), Metric.FAHRENHEIT);
+        calendar.addTemperature(date, temperature);
+
+        Day retrievedDay = calendar.getDay(date);
+        Temperature retrievedTemp = retrievedDay.getTemperature();
+
+        assertEquals(temperature, retrievedTemp);
```

Comment:
This is making the assumption that the stored temperature is the same object as the retrieved temperature. Better to compare the properties of the objects instead of the object instances.